index.js

```diff
@@ -100,30 +103,27 @@ for (let i = 0; i < datasets.length; i++) {
             }
         }
 
-        let bounding_box_area = bounding_box_length * bounding_box_height;
+        bounding_box_area = bounding_box_length * bounding_box_height;
 
         if (bounding_box_area / MAX_BOUNDING_BOX_AREA < 1) {
-            marker = L.polygon(bounding_box).addTo(map);
+            marker = L.polygon(bounding_box)
         }
 
     }
 
     var button_div = null;
     if (marker !== null) {
         
-        marker.on('click', function(e) {
-            toggleMetadata(i, true);
-            window.location.hash = "#" + datasets[i]["gmd:MD_Metadata"]["gmd:fileIdentifier"];
-        });
-        markers.push(marker);
+        let markers_length = markers.length;
 
-        let m = markers.length - 1;
+        markers.push(marker);
+        marker_size_and_index_pairs.push([bounding_box_area, markers_length]);
     
         button_div =   '<div class="m-2">' +
-                                `<button id="showBounds-${i}" class="btn btn-link" onclick="toggleBounds(${m});" style="display: none; z-index: 2000;">` +
+                                `<button id="showBounds-${i}" class="btn btn-link" onclick="toggleBounds(${markers_length});" style="display: none; z-index: 2000;">` +
```

Comment:
This might be correct, but why was it `markers.length - 1` (=`m`) before and is now `markers.length`?

Reply:
The assignment of m came after the marker was pushed before. I have now swapped the order to make things cleaner with the pushes to marker_size_and_index_pairs

Reply:
Ah right, I see. Thanks!